### PR TITLE
#38 frontend handling for expired jwt

### PR DESCRIPTION
**Pull request by Marlin-Noethig**
Added check for jwt expiration and automatic log out + some minor fixes.

Still get an error message when logged out automatically due to a state manipulation of a context while a routing is rendered. But works as intended anyway. Here is what the message looks like: 

***Edit: Error has been fixed in latest comment***

Bildschirmfoto 2022-05-30 um 19 12 37

### CODE REVIEW

File: .github/workflows/deploytoheroku.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
       - uses: actions/setup-java@v2
         with:
           java-version: '17'
-          distribution: 'adopt'
+          distribution: 'temurin'
```
Comment (Boltzmann) [inline .github/workflows/deploytoheroku.yml:34]: Why did you choose temurin? :curiosity:

File: frontend/src/routing/RequireAuth.tsx [hunk 2]
```diff
@@ -2,7 +2,10 @@ import {useContext} from "react";
 import {AuthContext} from "../context/AuthProvider";
 import {Navigate, Outlet} from "react-router-dom";
 
+
```
Comment (Boltzmann) [inline frontend/src/routing/RequireAuth.tsx:5]: Remove these two unnecessary lines. :justmy2cents:

**Review by Boltzmann — APPROVED** [review]
All is good. Only one question why temurin instead of openjdk, and one very small hint that two empty lines are unnecessary.

File: frontend/src/routing/RequireAuth.tsx [hunk 3]
```diff
@@ -2,7 +2,10 @@ import {useContext} from "react";
...
+
```
Comment (Marlin-Noethig) [inline frontend/src/routing/RequireAuth.tsx:6]:
```suggestion
```

**Pull request merged**